controllers/datafoundation.go

```diff
@@ -195,3 +227,55 @@ func (r *dataFoundationReconciler) setDeviceSetCount(deviceSet *ocsv1.StorageDev
 		deviceSet.Count = currDeviceSetCount
 	}
 }
+
+func (r *dataFoundationReconciler) reconcileCSV() error {
+	r.Log.Info("Reconciling OCS Operator CSV")
+	csvList := opv1a1.ClusterServiceVersionList{}
+	if err := r.list(&csvList); err != nil {
+		return fmt.Errorf("unable to list csv resources: %v", err)
+	}
+
+	for index := range csvList.Items {
+		csv := &csvList.Items[index]
+		if strings.HasPrefix(csv.Name, ocsOperatorName) {
+			if err := r.updateOCSCSV(csv); err != nil {
+				return fmt.Errorf("failed to update OCS CSV: %v", err)
+			}
```

Comment:
Why are we calling an update operation inside the loop? use the loop to find the CSV and then act on it after